go-controller/pkg/ovn/gress_policy.go

```diff
@@ -45,24 +45,54 @@ type portPolicy struct {
 	port     int32
 }
 
-func (pp *portPolicy) getL4Match() (string, error) {
+func (pp *portPolicy) getL4Match(protoFilters map[string]string) error {
 	if pp.protocol == TCP {
+		// If we already permit all TCP traffic, we don't need to add the ports.
+		// If there is an update that removes blanket tcp, then we'll reevaluate the
+		// matches. Same for the other protocols too
+		if protoFilters["tcp"] == "tcp" {
```

Comment:
if xxx["tcp"] == "tcp" looks confusing. Since the value is supposed to be L4 ports, and the keyword here is supposed to mean "all ports", it would be more clear to just use "all" as the keyword. With the keyword "all", this function can be very simple because the 3 branches for different protocols can be combined to one common logic.